padertorch/ops/losses/source_separation.py

```diff
@@ -105,18 +105,194 @@ def pit_loss(
+        (See the special implementation in this function for cross_entropy).
+        One good point is, that most used loss functions can be factorized.
+
+    Does not support batch dimension. Does not support PackedSequence.
+
+    Args:
+        estimate: Padded sequence. The speaker axis is specified with `axis`,
+            so the default shape is (T, K, F)
+        target: Padded sequence with the same shape as `estimate` (defaults
+            to (T, K, F))
+        loss_fn: Loss function to apply on each permutation. It must accept two
+            arguments (estimate and target) of the same shape that this function
+            receives the arguments.
+        axis: Speaker axis K. The permutation is applied along this axis. axis=-2
+            and an input shape of (T, K, F) corresponds to the old default
+            behaviour.
+
+    Examples:
+        >>> T, K, F = 4, 2, 5
+        >>> estimate, target = torch.ones(T, K, F), torch.zeros(T, K, F)
+        >>> pit_loss_from_pair_wise(pair_wise_loss(estimate, target, 1))
+        tensor(1.)
+
+        >>> T, K, F = 4, 2, 5
+        >>> estimate, target = torch.ones(T, K, F), torch.zeros(T, F, dtype=torch.int64)
+        >>> pit_loss_from_pair_wise(pair_wise_loss(estimate, target, 1, loss_fn=torch.nn.functional.cross_entropy), reduction='sum')
+        tensor(0.6931)
+        >>> pit_loss(estimate, target, 1, loss_fn=torch.nn.functional.cross_entropy)
+        tensor(0.6931)
+
+        >>> T, K, F = 4, 2, 5
+        >>> estimate, target = torch.ones(K, F, T), torch.zeros(K, F, T)
+        >>> pit_loss_from_pair_wise(pair_wise_loss(estimate, target, 0))
+        tensor(1.)
+
+        >>> T, K, F = 4, 2, 5
+        >>> estimate = torch.stack([torch.ones(F, T), torch.zeros(F, T)])
+        >>> target = estimate[(1, 0), :, :]
+        >>> pit_loss_from_pair_wise(pair_wise_loss(estimate, target, axis=0), return_permutation=True)
+        (tensor(0.), array([1, 0]))
+
+        >>> K = 5
+        >>> estimate, target = torch.ones(K), torch.zeros(K)
+        >>> pit_loss_from_pair_wise(pair_wise_loss(estimate, target, axis=0))
+        tensor(1.)
+
+        >>> A, B, K, C, F = 4, 5, 3, 100, 128
+        >>> estimate, target = torch.ones(A, B, K, C, F), torch.zeros(A, B, K, C, F)
+        >>> pit_loss_from_pair_wise(pair_wise_loss(estimate, target, axis=-3))
+        tensor(1.)
+    """
+    # axis = axis % estimate.ndimension()
+    sources = estimate.size()[axis]
+    assert sources < 30, f'Are you sure? sources={sources}'
+    if loss_fn in [torch.nn.functional.cross_entropy]:
+        estimate_shape = list(estimate.shape)
+        del estimate_shape[1]
+        assert estimate_shape == list(target.shape), (
+            f'{estimate.shape} (N, K, ...) does not match {target.shape} (N, ...)'
+        )
+    else:
+        assert estimate.size() == target.size(), (
+            f'{estimate.size()} != {target.size()}'
+        )
+
+    pair_wise_loss = []
```

Comment:
This is bad naming because the function has the same name. You could just call it `loss` or `pair_wise_losses` because it collects loss values of all pairs

Reply:
I removed this variable in this function (Was from an older version of the implementation)